pallets/operations/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![doc = include_str!("../README.md")]
+
+extern crate core;
+
+mod impls;
+mod traits;
+
+#[cfg(test)]
+mod tests;
+
+use frame_support::traits::{LockIdentifier, StorageVersion};
+
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+pub(crate) const LOG_TARGET: &str = "pallet-operations";
+// harcoding as those consts are not public in substrate
+pub(crate) const STAKING_ID: LockIdentifier = *b"staking ";
+pub(crate) const VESTING_ID: LockIdentifier = *b"vesting ";
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+#[pallet_doc("../README.md")]
+pub mod pallet {
+    use frame_support::pallet_prelude::*;
+    use frame_system::{ensure_signed, pallet_prelude::OriginFor};
+
+    use crate::{
+        traits::{AccountInfoProvider, BalancesProvider, NextKeysSessionProvider},
+        STORAGE_VERSION,
+    };
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        /// Something that provides information about an account's consumers counter
+        type AccountInfoProvider: AccountInfoProvider<AccountId = Self::AccountId, RefCount = u32>;
+        type BalancesProvider: BalancesProvider<AccountId = Self::AccountId>;
+        type NextKeysSessionProvider: NextKeysSessionProvider<AccountId = Self::AccountId>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// An account has fixed its consumers counter underflow
+        ConsumersUnderflowFixed { who: T::AccountId },
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// An account can have an underflow of a `consumers` counter.
+        /// Account categories that are impacted by this issue depends on a chain runtime,
+        /// but specifically for AlephNode runtime are as follows:
+        /// * `consumers`  == 0, `reserved`  > 0
+        /// * `consumers`  == 1, `balances.Locks` contain an entry with `id`  == `vesting`
+        /// * `consumers`  == 2, `balances.Locks` contain an entry with `id`  == `staking`
+        /// * `consumers`  == 3, `balances.Locks` contain entries with `id`  == `staking`
+        ///    and account id is in `session.nextKeys`
+        ///
+        ///	`fix_accounts_consumers_underflow` checks if the account falls into one of above
+        /// categories, and increase its `consumers` counter.
+        ///
+        /// - `origin`: Must be `Signed`.
+        /// - `who`: An account to be fixed
+        ///
+        #[pallet::call_index(0)]
+        // TODO: below weight is a maximum of DbReads ext can done, but it was not benchmarked
+        #[pallet::weight(
+        Weight::from_parts(0, 0)
```

Comment:
For the weight, let's just put here 2% of the whole block weight -- this is more than enough, so it's safe.

AFAIK a whole block is worth 1 AZERO, so this will cost 0.02 AZERO, which is reasonable.

Reply:
Done: 

![image](https://github.com/Cardinal-Cryptography/aleph-node/assets/3909333/be932799-753f-4e2e-a19f-a659ae842c65)

Reply:
Just to make sure, is the fee ~0.02 :)?

Reply:
Here's an example of batching this call in a single block (25 calls). 50% block capacity, fee 0.2 AZERO.

![image](https://github.com/Cardinal-Cryptography/aleph-node/assets/3909333/d91af3d0-a5a3-45ff-9465-51cf6b148298)

Reply:
ie so one transaction will cost 0.008 AZERO, is it okay?

Reply:
Ok, It works -- actually:
- 10^12 weight = 1 AZERO
- 10^12 weight = 1 sec, reference time
- we allow a block to have 0.4 sec of weight measured in ref_time. 

So 2% of block = 0.02*0.4 = 0.008 AZERO.